packages/plugin-dom-editable/src/EventNormalizer.ts

```diff
@@ -1644,6 +1664,23 @@ export class EventNormalizer {
                 this._mousedownInEditable = false;
                 this._initialCaretPosition = undefined;
             }
+        } else if (ev.target instanceof Element) {
+            // When within a contenteditable element but in a non-editable
+            // context, prevent a collapsed selection by removing all ranges.
+            // TODO: remove them from the VDocument as well.
+            new Timeout<EventBatch>(() => {
+                const selection = this._getSelection();
+                const collapsed =
+                    selection.anchorNode === selection.focusNode &&
+                    selection.anchorOffset === selection.focusOffset;
+                const target = this._getElement(selection.focusNode);
+                if (collapsed && !!target.closest('[contentEditable=true]')) {
+                    document.getSelection().removeAllRanges();
+                }
+                return {
+                    actions: [],
+                };
+            });
```

Comment:
Here is a problem as mentioned in a comment above.
The normalizer is not meant to correct the DOM. Only normalizing event of the DOM editable.
So `document.getSelection().removeAllRanges();` is forbidden in the Normalizer.
The Normalizer should send the event, our vDocument correct the change if necessary.

Do you understand why do we use the `Timeout` mechanism?
In this particular case, it is useless to use the `Timeout` class as it reduce to a call to a simple `setTimeout(fn, 0)`.

Even worse, the selection can be removed at the wrong time because of it's "random" call time (dependent of the system performance).
And it will never send the EventBatch as you are not calling `this._triggerEventBatch`.

Reply:
You are using the `Timeout` class but not it's usage. You can read the comment above `_pointerSelectionTimeout` to grasp the reason of it's existance.
The mechanism is to:
- make the function timeout to wait for the selection to be correct (you did this properly)
- save the callback that need to be called
- in case performance issue arise, the stack of event can be too big, manually trigger the callback at the right time (otherwise it would be called much later and the selection will be wrong)
- if the callback has been called properly, it will not trigger the timeout

So use the same mechanics, you should do something like this:
```typescript
this._pointerSelectionTimeout = new Timeout<EventBatch>(() => {
                const selection = this._getSelection();
                const collapsed =
                    selection.anchorNode === selection.focusNode &&
                    selection.anchorOffset === selection.focusOffset;
                const target = this._getElement(selection.focusNode);
                if (collapsed && !!target.closest('[contentEditable=true]')) {
                    document.getSelection().removeAllRanges();
                }
                return {
                    actions: [],
                };
});
this._triggerEventBatch(this._pointerSelectionTimeout.promise);
```

Without that the sequence of event when the system lags:
key: callback1
click: callback2
key: callback3

would be interpreted as
key: callback1
key: callback3
click: callback2

Hope it make sense to you.

Reply:
I don't get it. If someone click on a target that is in contenteditable:
- the normaliser send "setSelection" to the editor
- the editor process that the selection is not in an editable
- the editor discard the selection
- the editor rerender the selection

Reply:
I find that risky and out of scope. I'm open to discussion though for sure but I don't consider this a priority in the current state of affairs.

Reply:
1) Changing the logic about who revert the selection: can be done in another commit.
2) The timeout: cannot be skipped: use my code to trigger the event at the right moment.

Reply:
Done.